gears/src/x/types/delegation.rs

```diff
@@ -0,0 +1,9 @@
+use cosmwasm_std::Decimal256;
+
+use crate::types::address::{AccAddress, ValAddress};
+
+pub trait StakingDelegation {
+    fn delegator(&self) -> &AccAddress;
+    fn validator(&self) -> &ValAddress;
+    fn shares(&self) -> &Decimal256;
```

Comment:
So the shares can be fractional? Is this what they use in Cosmos?

Reply:
This is what we have